p_steps/step_12_1_create_scri_dataset.R

```diff
@@ -0,0 +1,293 @@
+#               parameters: in 07_scri_inputs.R
+#               output: g_output/scri/scri_input_[brand]_[outcome]_flowchart.txt
+#                       g_intermediate/scri/scri_input_[brand]_[outcome]_input.csv               
+
+# Housekeeping  -----------------------------------------------------------
+# install and load packages 
+if (!require("pacman")) install.packages("pacman")
+pacman::p_load(packages, character.only = TRUE)
+
+# ensure required folders are created 
+dir.create(file.path("./g_intermediate/scri"), showWarnings = FALSE, recursive = TRUE)
+dir.create(file.path("./g_output/scri"), showWarnings = FALSE, recursive = TRUE)
+
+# Functions for Cohort Creation -------------------------------------------
+# note - one thing to change in these I think is the hard coded data name, which is awkward 
+
+#' FUNCTION 1. scri flowchart
+#' takes the an input dataset and outputs a flowchart as txt, per outcome and vaccine
+#' 
+#' @param brand = vaccine brand, as quoted character
+#' @param outcome = outcome, as quoted character 
+#' @param outcome_name = outcome display name, as quoted character
+
+scri_flowchart <- function(brand, outcome, outcome_name) {
+  
+  # create required variables to determine study period 
+  scri_data_extract <- scri_data_extract %>% 
+    mutate(start = date_vax1 - 90, 
+           riskd1 = date_vax1 + 28, 
+           riskd2 = date_vax2 + 28) %>% 
+    # calculate the end of the risk period taking into account both doses
+    mutate(end_risk = case_when(!is.na(date_vax2) ~ riskd2, 
+                                TRUE ~ riskd1)) %>% 
+    rowwise() %>% 
+    # create study end date 
+    mutate(end = min(end_risk, study_exit_date, death_date, na.rm = T)) %>% 
+    ungroup()
+  
+  total <- scri_data_extract %>% 
+    nrow() 
+  
+  step1 <- scri_data_extract %>%
+    filter(!is.na(sex)) %>% 
+    nrow()
+  
+  step2 <- scri_data_extract %>%
+    filter(!is.na(sex), 
+           !is.na(age_at_study_entry)) %>% 
+    nrow()
+  
+  step3 <- scri_data_extract %>%
+    filter(!is.na(sex), 
+           !is.na(age_at_study_entry), 
+           !is.na(date_vax1)) %>% 
+    nrow()
+  
+  step4 <- scri_data_extract %>% 
+    filter(!is.na(sex), 
+           !is.na(age_at_study_entry), 
+           !is.na(date_vax1), 
+           study_entry_date <= date_vax1 & date_vax1 <= study_exit_date) %>% 
+    nrow() 
+  
+  step5 <- scri_data_extract %>%
+    filter(!is.na(sex), 
+           !is.na(age_at_study_entry), 
+           !is.na(date_vax1), 
+           study_entry_date <= date_vax1 & date_vax1 <= study_exit_date, 
+           type_vax_1 == {{brand}}) %>% 
+    nrow()
+  
+  step6 <- scri_data_extract %>% 
+    filter(!is.na(sex), 
+           !is.na(age_at_study_entry), 
+           !is.na(date_vax1), 
+           study_entry_date <= date_vax1 & date_vax1 <= study_exit_date, 
+           type_vax_1 == {{brand}}, 
+           !is.na({{outcome}})) %>% 
+    nrow()
+  
```

Comment:
I run the command in line 292 (pmap(scri_input_grid, scri_flowchart))in my computer and it gives me the following error: 
Error: Problem with `filter()` input `..6`.
ℹ Input `..6` is `!is.na(c("myocarditis_date", "pericarditis_date"))`.
x Input `..6` must be of size 10000 or 1, not size 2.
Run `rlang::last_error()` to see where the error occurred.
In addition: Warning messages:
1: In type_vax_1 == ~c("Pfizer", "AstraZeneca", "Moderna") :
  longer object length is not a multiple of shorter object length
2: In type_vax_1 == ~c("Pfizer", "AstraZeneca", "Moderna") :

It seems that this error starts from step5 above. Perhaps the simplification that Sophie suggested above will solve this?

Reply:
This seems as if, at some point in your data, you end up with no myocarditis and pericarditis dates. Could that be correct? I do not get this error, which makes me think it's a consequence of the data in some way, and it seems to be a type error, which makes me think it's maybe seeing NAs somewhere. Are you running the script from start to finish?